src/pngwriter.cc

```diff
@@ -787,23 +795,24 @@ int pngwriter::read(int x, int y, int colour) const
 
 	if(bit_depth_ == 8)
 	  {
+	     int const scale8To16Bit = 257;   // x/255*65535 = x*257
```

Comment:
This is a normal division slash, which well in my opinion and in the opinion of the C++ language only affects the very next number, not the rest of the line. I didn't want to write your version, because I imagine it this way:
  1. scale down by `/255` to reach `[0,1.0]`
  2. scale up by `65535`
You self-assigned this pull request, does that mean you will do the changes yourself or does that only mean you will peer review it?